criticality_score/run.py

```diff
@@ -541,6 +543,8 @@ def get_repository(url):
         repo_url_encoded = urllib.parse.quote_plus(repo_url)
         try:
             repo = token_obj.projects.get(repo_url_encoded)
+            if len(repo.commits.list()) == 0:
```

Comment:
How slow is this call ?

Reply:
How about making these "get commits" calls in advance?

I added "last_commit" property to Repository class. Before initializing the repo, we make these "get commits" calls, do the validations, if it's all good, then create the repo by passing "last commit" as a parameter.

Since we only pass "last_commit" as a parameter, it feels bit strange but with this approach, we still do the validations and not making additional calls.

Let me know what you think.